nsq/reader.py

```diff
@@ -677,7 +669,7 @@ def _redistribute_rdy_state(self):
             # We also don't attempt to avoid the connections who previously might have had RDY 1
             # because it would be overly complicated and not actually worth it (ie. given enough
             # redistribution rounds it doesn't matter).
-            possible_conns = list(self.conns.values())
+            possible_conns = [c for c in conns if not (c.in_flight or c.rdy)]
```

Comment:
Consider a case where `max_in_flight` is less than the connection count and a task takes longer than `low_rdy_idle_timeout`.   Then we'll set `RDY 0` on a connection even while it's still processing a message (https://github.com/nsqio/pynsq/blob/master/nsq/reader.py#L664).  So we can end up in a situation where `total_rdy` is strictly less than the total count of in-flight messages.  If we go from from `RDY 0` to `RDY 1` on a connection that has messages available for delivery, we guarantee a max-in-flight violation.